### PR TITLE
Price Impact Warning

### PR DESCRIPTION
**Pull request by walmat**
Fixes APP-1307, APP-1366, APP-1362

### CODE REVIEW

File: src/__swaps__/screens/Swap/components/PriceImpactWarning.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+export const PriceImpactWarning = () => {
+  const { AnimatedSwapStyles, PriceImpactWarning } = useSwapContext();
+
+  // TODO: i18n is not liking this...
+  const warningText = useDerivedValue(() => {
+    if (PriceImpactWarning.value.type === SwapPriceImpactType.none) return '';
+    return i18n.t(i18n.l.exchange.price_impact.you_are_losing, {
+      impactDisplay: PriceImpactWarning.value.impactDisplay,
+    });
+  });
```
Comment (walmat) [inline src/__swaps__/screens/Swap/components/PriceImpactWarning.tsx:18]: is there a way to use i18n inside of a derived value here? @christianbaroni

Comment (walmat) [reply src/__swaps__/screens/Swap/components/PriceImpactWarning.tsx:18]: Screenshot 2024-04-17 at 1 14 27 PM

File: src/__swaps__/screens/Swap/components/PriceImpactWarning.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+          >
+            <TextIcon color="orange" containerSize={20} size="15pt" weight="heavy">
+              􀇿
+            </TextIcon>
+          </Box>
+          <AnimatedText align="center" color="orange" size="15pt" weight="heavy" text={warningText} />
+        </Inline>
+
+        <Text color="labelQuaternary" size="13pt" weight="bold">
+          {i18n.t(i18n.l.exchange.price_impact.smal_market_try_smaller_amount)}
```
Comment (dereknelson) [inline src/__swaps__/screens/Swap/components/PriceImpactWarning.tsx:48]: we should add some logic for when price impact shows 100% (which means we have no data) and we can also warn them that they should be happy with the quote https://github.com/rainbow-me/rainbow/pull/5635/files#diff-1cd0ecae5857bb98974614524f7c7c124225df7e11fc1b4b22d176dc2f9abce6R571

**Review by dereknelson — CHANGES_REQUESTED** [review]

File: src/__swaps__/screens/Swap/components/PriceImpactWarning.tsx (hunk 2 above)
Comment (walmat) [reply src/__swaps__/screens/Swap/components/PriceImpactWarning.tsx:38]: added a follow-up ticket: https://linear.app/rainbow/issue/APP-1366/handle-price-impact-case-where-we-have-no-price-data

**Review by skylarbarrera — APPROVED** [review]

**Pull request merged**